src/AsyncTCP.cpp

```diff
@@ -925,6 +925,10 @@ int8_t AsyncClient::abort() {
     _tcp_abort(&_pcb, this);
     // _pcb is now NULL
   }
+  if (_discard_cb) {
+    // _pcb was closed here
+    _discard_cb(_discard_cb_arg, this);
+  }
```

Comment:
@willmmiles: no sure if this block should be within the `if (_pcb) {` block or not. If yes, then maybe the same is true for close also and the latter should be updated too ?

What do you think ?